File: exercises/week01/part02/Student.java
package part02;

public class Student {
  private String name;
  private String studentId;
  private int credits;

  public Student () {}

  public Student ( String name, String studentId ) {
    this.name = name;
    this.studentId = studentId;
  }

  //getters

  public String getName () { return this.name; }
  public String getStudentId () { return this.studentId; }
  public int getCredits () {  return this.credits; }

  //setters

  public void setName ( String name ) {
    this.name = name;
  }

  public void setStudentId ( String studentId ) {
    this.studentId = studentId;
  }

  public void setCredits ( int credits ) {
    this.credits = credits;
  }

  //public methods

  public void print () {
    var info = String.format("Name: %s\nStudent number: %s\nCredits: %s\n", 
      this.getName(),
      this.getStudentId(),
      this.getCredits()
    );

    System.out.println(info);
  }
}

File: exercises/week05/part02/App/student.cpp
#include "student.h"

Student::Student(void) {}

Student::Student(const std::string& name, const std::string& student_id)
  : _name(name), _student_id(student_id), _credits(0) {}

Student::~Student() {}

//getters

std::string Student::name(void) const { return this->_name; }

std::string Student::student_id(void) const { return this->_student_id; }

int Student::credits(void) const { return this->_credits; }

Address Student::address(void) const { return this->_address; }

//setters

void Student::set_name(const std::string& name) {
  this->_name = name;
}

void Student::set_student_id(const std::string& student_id) {
  this->_student_id = student_id;
}

void Student::set_credits(const int& credits) {
  this->_credits = credits;
}

void Student::set_address(const std::string &street, const std::string &zip, const std::string &municipality) {
    this->_address.set_address(street, zip, municipality);
}
